test/models/profile_test.dart

```diff
@@ -0,0 +1,185 @@
+      Profile profile = ProfileFactory().generateFake(
+        name: NullableParameter(null),
+        surname: NullableParameter(null),
+      );
+      expect(profile.fullName, '');
+    });
+
+    test('returns surname if name is not set', () async {
+      final profile = ProfileFactory().generateFake(
+        name: NullableParameter(null),
+      );
+      expect(profile.fullName, profile.surname);
+    });
+
+    test('returns name if surname is not set', () async {
+      final profile = ProfileFactory().generateFake(
+        surname: NullableParameter(null),
+      );
+      expect(profile.fullName, profile.name);
+    });
+
+    test('returns name and surname if both are set', () async {
+      final profile = ProfileFactory().generateFake(
+        name: NullableParameter('name'),
+        surname: NullableParameter('surname'),
+      );
+      expect(profile.fullName, 'surname name');
+    });
+  });
+
+  group('isCurrentUser', () {
+    test('returns true if id is equal to current profile id', () async {
+      final profile = ProfileFactory().generateFake();
+      final currentProfile = ProfileFactory().generateFake(
+        id: profile.id,
+      );
+      SupabaseManager.setCurrentProfile(currentProfile);
+      expect(profile.isCurrentUser, true);
+    });
+
+    test('returns false if id is not equal to current profile id', () async {
+      final profile = ProfileFactory().generateFake();
+      final currentProfile = ProfileFactory().generateFake(id: profile.id! + 1);
+      SupabaseManager.setCurrentProfile(currentProfile);
+      expect(profile.isCurrentUser, false);
+    });
+  });
+
+  group('Profile.fromJson', () {
+    test('parses a profile from json', () async {
+      Map<String, dynamic> json = {
+        "id": 1,
+        "created_at": "2021-01-01T00:00:00.000Z",
+        "username": "username",
+        "email": "email",
+        "description": "description",
+        "birth_date": "2021-01-01T00:00:00.000Z",
+        "surname": "surname",
+        "name": "name",
+        "gender": 0,
+        "avatar_url": "avatar_url",
+      };
+      Profile profile = Profile.fromJson(json);
+      expect(profile.id, json["id"]);
+      expect(profile.createdAt, DateTime.parse(json["created_at"]));
+      expect(profile.username, json["username"]);
+      expect(profile.email, json["email"]);
+      expect(profile.description, json["description"]);
+      expect(profile.birthDate, DateTime.parse(json["birth_date"]));
+      expect(profile.surname, json["surname"]);
+      expect(profile.name, json["name"]);
+      expect(profile.gender, Gender.male);
+      expect(profile.avatarUrl, json["avatar_url"]);
+    });
+
+    test('can handle associated models', () async {
+      Map<String, dynamic> json = {
+        "id": 1,
+        "created_at": "2021-01-01T00:00:00.000Z",
+        "username": "username",
+        "email": "email",
+        "reviews_received": ReviewFactory().generateFakeJsonList(length: 3),
```

Comment:
does this work if the reviews_received == null same for the others?

Reply:
Do you mean if we used NullableParameter to set them?

Reply:
yes if thats the way to do that

Reply:
Okay sorry for my confusion. That would be the way if we used the factory here. But we decided to hard-code the json for these specific tests so we can test independently from the correctness of the `toJson` method.

As to your original question: In this case, null is the default. So this is covered in the test case one above, 'parses a profile from json' where those values are not set.

Reply:
yeah that's right my mistake